components/scream/docs/source-tree.md

```diff
@@ -0,0 +1,63 @@
+# SCREAM's Source Tree
+
+All SCREAM-specific code can be found in `components/scream` within the
+[SCREAM repo](https://github.com/E3SM-Project/scream). Here's how things are
+organized:
+
++ `bin`: Miscellaneous tools for helping with the build process. Currently, this
+  directory only contains a template for generating the `scream_mpicxx` compiler
+  script.
++ `cime_config`: Tools and XML files for integrating SCREAM with E3SM via the
+  CIME framework.
++ `cmake`: CMake functions and macros used by the configuration/build system.
++ `data`: Data files used by our tests.
++ `docs`: Documentation for the SCREAM project, including design documents,
+  instructions for building and testing SCREAM, and this document.
++ `extern`: Source for certain lightweight third-party libraries, embedded
+  directly into the repo (and not imported as submodules).
++ `scripts`: Miscellaneous scripts that implement workflows for running tests
+  and analyzing performance.
++ `src`: All C++ source code (and any bridges to Fortran) for SCREAM are stored
+  here. We describe the contents of this directory in greater detail below.
++ `tests`: Implements standalone, end-to-end tests for various SCREAM
+  components (RRTMG, HOMME, P3, SHOC, etc).
+
+In addition, you'll notice the following files in `components/scream`:
+
++ `CMakeLists.txt`: The CMake file that defines SCREAM's configuration/build
+  system.
++ `CTestConfig.cmake`: This CTest file contains parameters that determine how
+  our test results are reported to the [E3SM CDash Site](http://my.cdash.org/submit.php?project=E3SM).
++ `README.md`: SCREAM's top-level README file, which describes the project and
+  its purpose.
+
+## The `src` Directory
+
+Herein lіes the source code for SCREAM. Broadly, here's where things are:
+
++ `control`: Contains the atmosphere driver and basic tests for it.
++ `dynamics`: Here's where HOMME lives within SCREAM, along with code for
+  interfacing with it using SCREAM's data structures.
++ `interface`: Glue code for embedding SCREAM within E3SM as an atmosphere
+  process.
++ `physics`: Source code for physics-related atmospheric processes, including
+  + `p3`: The C++/Kokkos implementation of P3 microphysics within SCREAM.
+  + `shoc`: The C++/Kokkos implementation of SHOC macrophysics within SCREAM.
+  + `rrtmgp`: A stub for the radiation processes as represented in SCREAM.
+  + `common`: Utilities and data structures common to these processes.
```

Comment:
I still want to rename 'common' -> 'share' to match terminology used elsewhere in our directory structure. Again, not for this PR.

Reply:
I'm not sure how we landed on that name, but we might have picked common precisely to avoid confusion with the other (main) share folder. Personally, I do think 'common' (or 'generic') is better than share, in this context. But I don't have a very strong opinion about it.

Reply:
Again, let's leave things as-is and see if divine inspiration strikes us in the next few weeks about what to do.